wvlet-lang/src/main/scala/wvlet/lang/compiler/parser/SqlParser.scala

```diff
@@ -1098,6 +1098,23 @@ class SqlParser(unit: CompilationUnit, isContextUnit: Boolean = false) extends L
           nextCase
           consume(SqlToken.END)
           CaseExpr(target, cases.result(), elseClause, spanFrom(t))
+        case SqlToken.IF =>
```

Comment:
`if` is just a function name in SQL. So add this logic inside premaryExpression parser  https://trino.io/docs/current/functions/conditional.html#if